packages/authgear-react-native/ios/AGAuthgearReactNative.m

```diff
@@ -112,41 +114,21 @@ + (BOOL)application:(UIApplication *)application
                   resolve:(RCTPromiseResolveBlock)resolve
                    reject:(RCTPromiseRejectBlock)reject)
 {
-    if (@available(iOS 12.0, *) && !prefersSFSafariViewController) {
-        self.asSession = [[ASWebAuthenticationSession alloc] initWithURL:url
-                                                       callbackURLScheme:nil
-                                                       completionHandler:^(NSURL *url, NSError *error) {
-            self.asSession = nil;
-        }];
-        if (@available(iOS 13.0, *)) {
-            self.asSession.presentationContextProvider = self;
-        }
-        BOOL started = [self.asSession start];
-        if (!started) {
-            if (reject) {
-                reject(RCTErrorUnspecified, [NSString stringWithFormat:@"Unable to open URL: %@", url], nil);
-            }
-        } else {
-            if (resolve) {
-                resolve(nil);
-            }
-        }
-    } else if (@available(iOS 11.0, *) && !prefersSFSafariViewController) {
-        self.sfSession = [[SFAuthenticationSession alloc] initWithURL:url
-                                                    callbackURLScheme:nil
-                                                    completionHandler:^(NSURL *url, NSError *error){
-            self.sfSession = nil;
-        }];
-        BOOL started = [self.sfSession start];
-        if (!started) {
-            if (reject) {
-                reject(RCTErrorUnspecified, [NSString stringWithFormat:@"Unable to open URL: %@", url], nil);
-            }
-        } else {
-            if (resolve) {
-                resolve(nil);
-            }
-        }
+    if (!prefersSFSafariViewController) {
```

Comment:
Using SFSafariViewController would leak the app session cookie out of the app, so we use web view by default.

Reply:
Then I think our choice should be WKWebView and ASWebAuthenticationSession. SFSafariViewController should never be used.

Reply:
How about always use web view (i.e. remove the prefer safari option)? Using ASWebAuthenticationSession has incorrect semantics, since we're not doing authentication.

Reply:
For `authorize()`, we allow choosing between ASWebAuthenticationSession and WKWebView.
For `openURL`, we always use WKWebView.

This seems the simplest approach.